providers/storage/storageDocStore.js

```diff
@@ -47,13 +47,18 @@ class AzureStorageDocStore {
       blobMetadata.extra = JSON.stringify(document._metadata.extra);
     }
     const options = { metadata: blobMetadata, contentSettings: { contentType: 'application/json' } };
-    this.service.createBlockBlobFromText(this.name, blobName, text, options, (error, result, response) => {
-      if (error) {
+    const dataStream = new Readable();
+    dataStream.push(JSON.stringify(document));
+    dataStream.push(null);
+    dataStream
+      .pipe(this.service.createWriteStreamToBlockBlob(this.name, blobName, options))
+      .on('error', (error) => {
         return deferred.reject(error);
-      }
-      memoryCache.put(document._metadata.url, { etag: document._metadata.etag, document: document }, this.options.ttl);
-      deferred.resolve(blobName);
-    });
+      })
+      .on('finish', () => {
+        memoryCache.put(document._metadata.url, { etag: document._metadata.etag, document: document }, this.options.ttl);
```

Comment:
If we're supporting files over 64 Mb should we be concerned that we're putting documents into a memory cache? Should we not cache files over a certain size?

Reply:
Good point. I'll add a condition for that.

Reply:
As discussed, I've removed memory caching for upsert, get, and etag.